File: parser/src/main/java/io/ticofab/androidgpxparser/parser/domain/Copyright.java
package io.ticofab.androidgpxparser.parser.domain;

public class Copyright {

    private final String mAuthor;
    private final Integer mYear;
    private final String mLicense;

    private Copyright(Builder builder) {
        mAuthor = builder.mAuthor;
        mYear = builder.mYear;
        mLicense = builder.mLicense;
    }

    public String getAuthor() {
        return mAuthor;
    }

    public Integer getYear() {
        return mYear;
    }

    public String getLicense() {
        return mLicense;
    }

    public static class Builder {
        private String mAuthor;
        private Integer mYear;
        private String mLicense;

        public Builder setAuthor(String author) {
            mAuthor = author;
            return this;
        }

        public Builder setYear(Integer year) {
            mYear = year;
            return this;
        }

        public Builder setLicense(String license) {
            mLicense = license;
            return this;
        }

        public Copyright build() {
            return new Copyright(this);
        }
    }
}

File: parser/src/main/kotlin/io/ticofab/androidgpxparser/parser/domain/Copyright.kt
package io.ticofab.androidgpxparser.parser.domain

class Copyright private constructor(builder: Builder) {
    val author: String?
    val year: Int?
    val license: String?

    class Builder {
        var author: String? = null
            private set
        var year: Int? = null
            private set
        var license: String? = null
            private set

        fun setAuthor(author: String?): Builder {
            this.author = author
            return this
        }

        fun setYear(year: Int?): Builder {
            this.year = year
            return this
        }

        fun setLicense(license: String?): Builder {
            this.license = license
            return this
        }

        fun build(): Copyright {
            return Copyright(this)
        }
    }

    init {
        author = builder.author
        year = builder.year
        license = builder.license
    }
}